### PR TITLE
Lets you wear cigarettes on your ears

### PR DESCRIPTION
**Pull request by JMWTurner**
earcig

who needs radio headsets anyway

works with lit cigarettes as well, but not with blunts, pipes, or cigars.

### CODE REVIEW

**Comment by Exxion** [comment]
matchear dont

**Comment by ghost** [comment]
this is the lame way to hold a cig in your ear.
everyone knows the patrician way is to stick the filter in.
(two ear slots when?)

**Comment by JMWTurner** [comment]
revision 1
- the cig overlay now updates when you light it up when it's on your ear
- added a bit of smoke for when wearing a lit cigarette while facing right

@whorena >two ear slots when
That's a good idea. someone should code it.

**Comment by Exxion** [comment]
Please tell me a lit cigarette on your ear can ignite plasma

**Comment by JMWTurner** [comment]
@Exxion of course
yep

**Comment by gbasood** [comment]
station where only your bosses and sec have radio headsets when

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -329,6 +330,8 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 	if(ismob(loc))
 		var/mob/M = loc
 		M.update_inv_wear_mask(0)
+		if(ishuman(loc))
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/cigs_lighters.dm]: There's no need for the check here (and if there was, you should be using M, not loc, for consistency). update_inv_ears is defined at the mob level.

**Comment by ComicIronic** [comment]
It's a great thing to see how smooth content additions can be in good systems.

**Comment by JMWTurner** [comment]
Isn't it

**Comment by Kurfursten** [comment]
There is still skeletoncode from when N3X attempted (failed?) to go to 2 ear slots.

**Comment by gbasood** [comment]
if only we weren't stuck with dumb UI I might actually be for that

**Comment by PJB3005** [comment]
Nobody is adding any fucking slots until we rework inventory code.

**Comment by clusterfack** [comment]
Rework inventory code he says :joy:

**Pull request merged**